### ISSUE TITLE
[Feature] No line "wrap/word" wrap

### ISSUE THREAD

**Issue by TimerWolf** [go-gitea/gitea#27277]
### Feature Description

I think it is really strange that Gitea not have this feature when you read the source of the file or commit in the respiratory.

My opinion is that it is a lot more easy to fallow the code if not the browser cut of the lines where it not should be.

It should as well be very easy to implant.

**Comment by TimerWolf** [comment]
Would it not be smarter to have an icon for it in the source view to access
it easy ?

If you would like to turn it "on / off" in my case it would not matter but
in the same time it does not feel like a feature that should be hidden.

Den tis 26 sep. 2023 16:45silverwind ***@***.***> skrev:

**Comment by silverwind** [comment]
Possibly, but I'd say it has to go into a menu, same row as file path. Diff view already has a such a "..." menu, so just add it there.

**Comment by TimerWolf** [comment]
Maybe smart, something that is really smart is no matter where that setting
will placed it does remember your choice so you not have to reset it
everytime.

Den tis 26 sep. 2023 19:55silverwind ***@***.***> skrev:

**Comment by silverwind** [comment]
Yes, remember in localStorage, or even better in server-stored user settings.

**Comment by TimerWolf** [comment]
Yeh, I never liked localStorage for this type of things but It can come
very handy to other things.

Everything in softwares like this should be server-side as in my case I
always empty everything every time I close my web browsers an think that
others made do like this as well.

It should not depend on browser ether, you should just need to set it once
and know the setting is there no matter what "browser / computer" you use.

Den tis 26 sep. 2023 20:16silverwind ***@***.***> skrev:

**Comment by silverwind** [comment]
Yes, I think it's just a matter of hooking up `GetUserSetting` and `SetUserSetting` from the backend, to my knowledge, no UI feature has done so far so they all use localStorage, but should be migrated.

**Comment by TimerWolf** [comment]
Sounds nice with someone that know how to so this, as I do not know anything about GO coding, hope you find some time to implant this improvment.

**Comment by wolftune** [comment]
is this overlap or duplicate with https://github.com/go-gitea/gitea/issues/22003 ?
